gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/avro/HiveAvroSerDeManager.java

```diff
@@ -79,7 +85,9 @@ public HiveAvroSerDeManager(State props) throws IOException {
     }
 
     this.useSchemaFile = props.getPropAsBoolean(USE_SCHEMA_FILE, DEFAULT_USE_SCHEMA_FILE);
+    this.useSchemaTempFile = props.getPropAsBoolean(USE_SCHEMA_TEMP_FILE, DEFAULT_USE_SCHEMA_TEMP_FILE);
     this.schemaFileName = props.getProp(SCHEMA_FILE_NAME, DEFAULT_SCHEMA_FILE_NAME);
+    this.schemaTempFileName = props.getProp(SCHEMA_TEMP_FILE_NAME, DEFAULT_SCHEMA_TEMP_FILE_NAME);
```

Comment:
what is the parent directory for this temp schema file? can it be same over or between executions, if so should we add a UUID to the name to avoid it being overwritten?

Reply:
The temp file is placed in the same directory as the schema file. It is only temporary and won't exist after being renamed to the final schema file name.